### DISCUSSION TITLE
Using Lego Spike Hub's FollowLine Question

### DISCUSSION THREAD

**Discussion by hashen0608** [pybricks/support#1601] [Q&A]
Hello everyone, I am using PyBricks and currently working on programming a line-following robot using the Spike robot.

Previously, I used Lego's programming software, and with the same approach, I was able to program the robot to follow lines. However, I'm encountering difficulties doing the same in PyBricks. I'm wondering if there's something I might not be doing correctly.

This is the program I wrote in the Lego software: it can run successfully.
![image](https://github.com/pybricks/support/assets/167075380/26149bfc-1195-451a-b5d3-f0e1d8d9ca0e)

This is the program I wrote in PyBricks: the program cannot run.
![image](https://github.com/pybricks/support/assets/167075380/088340bb-409b-403b-b508-09ef5a281ed1)

thanks

hashen

**Comment by laurensvalk** [comment]
Thanks for taking the time to ask it here. Here's one way to do it:

![image](https://github.com/pybricks/support/assets/12326241/bdf889a6-d18a-40b2-a057-f8019cc88f02)

Notice that you can conveniently use the drive base blocks for this. You don't have to give separate speeds for left and right, but you can specify how fast to turn.
